File: Javascript/Arrays & Hashing/Longest-Consecutive-Sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const longestConsecutive = (nums) => {
  let set = new Set(nums);

  let longestConsecutive = 0;

  for (let number of nums) {
    if (!set.has(number - 1)) {
      let current = 0;
      while (set.has(number + current)) {
        current++;
      }

      longestConsecutive = Math.max(current, longestConsecutive);
    }
  }

  return longestConsecutive;
};

File: Go/Arrays & Hashing/Longest-Consecutive-Sequence.go
package main

import "math"

func main() {

}

func longestConsecutive(nums []int) int {
	set := map[int]bool{}

	var longest float64
	for _, num := range nums {
		set[num] = true
	}

	for _, num := range nums {
		if !set[num - 1] {
			current := 0

			for set[num+current] {
				current++
			}

			longest = math.Max(float64(current), float64(longest))
		}
	}

	return int(longest)
}